core/tests/protractor_utils/TopicsAndSkillsDashboardPage.js

```diff
@@ -383,10 +389,11 @@ var TopicsAndSkillsDashboardPage = function() {
   };
 
   this.expectNumberOfTopicsToBe = async function(number) {
-    let topicsTableIsPresent = await topicsTable.isPresent();
-    if (topicsTableIsPresent) {
-      expect(await topicsListItems.count()).toBe(number);
+    if (number > 0) {
+      await waitFor.visibilityOf(
+        topicsListItems, 'Topics table taking too long to appear');
     }
+    expect(await topicsListItems.count()).toBe(number);
```

Comment:
This doesn't handle the case where number is 0, so no `topicsListItems` exist. In this case, we should check that `topicsListItems` is not present instead of counting.

Reply:
@U8NWXD I'm getting confused in this part, will you please guide me? What I can think of doing is
```
  this.expectNumberOfTopicsToBe = async function(number) {
    if (number > 0) {
      await waitFor.visibilityOf(
        topicsListItems, 'Topics table taking too long to appear');
      expect(await topicsListItems.count()).toBe(number);
    }
    expect(await topicsListItems.isPresent());
  };
```
Thanks

Reply:
Almost! The topics table is only present when topics are present, so in the case when `number` is 0, we should check that the topics table is *not* present. For example:

```
  this.expectNumberOfTopicsToBe = async function(number) {
    if (number > 0) {
      await waitFor.visibilityOf(
        topicsListItems, 'Topics table taking too long to appear');
      expect(await topicsListItems.count()).toBe(number);
    } else {
      expect(await topicsTable.isPresent()).toBe(false);
    }
  };
```

Reply:
Thanks